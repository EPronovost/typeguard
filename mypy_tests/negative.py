from typeguard import check_argument_types, check_return_type, typechecked


@typechecked
def foo(x: int) -> int:
    return x + 1


def arg_type(x: int) -> str:
    return check_argument_types()  # Error


def ret_type() -> str:
    return check_return_type(False)  # Error


_ = arg_type(foo)  # Error
