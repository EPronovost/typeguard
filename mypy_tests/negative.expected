negative.py:11: error: Incompatible return value type (got "str", expected "int")
negative.py:15: error: Incompatible return value type (got "bool", expected "str")
negative.py:19: error: Incompatible return value type (got "bool", expected "str")
negative.py:22: error: Argument 1 to "arg_type" has incompatible type "Callable[..., Any]"; expected "int"
Found 4 errors in 1 file (checked 1 source file)
